### ISSUE TITLE
composer 2.3 not compatible

### ISSUE THREAD

**Issue by demeritcowboy** [civicrm/composer-compile-plugin#19]
It seems to have made a change so that eventname can't be null for EventDispatcher. Admittedly 2.3 was just released today, but civi download fails at the compilation task.

```
Error: Argument 1 passed to Composer\EventDispatcher\Event::__construct() must be of the type string, null given, called in .../vendor/civicrm/composer-compile-plugin/src/Event/CompileTaskEvent.php on line 55
In Event.php line 49:
                                                                               
  [TypeError]                                                                  
  Argument 1 passed to Composer\EventDispatcher\Event::__construct() must be of the type string, null given, called in .../vendor/civicrm/composer-compile-plugin/src/Event/CompileTaskEvent.php on line 55          
                                                                               

Exception trace:
  at phar:///usr/local/bin/composer/src/Composer/EventDispatcher/Event.php:49
 Composer\EventDispatcher\Event->__construct() at .../vendor/civicrm/composer-compile-plugin/src/Event/CompileTaskEvent.php:55
 Civi\CompilePlugin\Event\CompileTaskEvent->__construct() at .../vendor/civicrm/composer-compile-plugin/src/TaskRunner.php:205
 Civi\CompilePlugin\TaskRunner->runTask() at .../vendor/civicrm/composer-compile-plugin/src/TaskRunner.php:169
 Civi\CompilePlugin\TaskRunner->run() at .../vendor/civicrm/composer-compile-plugin/src/TaskRunner.php:74
 Civi\CompilePlugin\TaskRunner->runDefault() at .../vendor/civicrm/composer-compile-plugin/src/Command/CompileCommand.php:75
 Civi\CompilePlugin\Command\CompileCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:298
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:1015
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:299
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:334
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:171
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:130
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:83
 require() at /usr/local/bin/composer:29
```

**Comment by herbdool** [comment]
I'm glad to have found this issue! Just ran into this issue when creating a build on tugboat.qa. I can confirm that there's still a bug here: `Return value of "Civi\CompilePlugin\Command\CompileCommand::execute()" must be of the t  
  ype int, "null" returned.`.

**Comment by totten** [comment]
@demeritcowboy @seamuslee001 @herbdool - The updates have been published as v0.18.

**Comment by demeritcowboy** [comment]
Thanks!
